maps_dashboards/public/components/layer_config/document_layer_source.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect, useState } from 'react';
+import { EuiComboBox, EuiFlexItem, EuiFormLabel, EuiFlexGrid, EuiFieldNumber } from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { IndexPattern, IndexPatternField } from '../../../../../src/plugins/data/public';
+import { useOpenSearchDashboards } from '../../../../../src/plugins/opensearch_dashboards_react/public';
+import { MapServices } from '../../types';
+import { DocumentLayerSpecification } from '../../model/mapLayerType';
+
+interface Props {
+  setSelectedLayerConfig: Function;
+  selectedLayerConfig: DocumentLayerSpecification;
+}
+
+export const DocumentLayerSource = ({ setSelectedLayerConfig, selectedLayerConfig }: Props) => {
+  const {
+    services: {
+      savedObjects: { client: savedObjectsClient },
+      data: {
+        ui: { IndexPatternSelect },
+        indexPatterns,
+      },
+    },
+  } = useOpenSearchDashboards<MapServices>();
+  const [indexPattern, setIndexPattern] = useState<IndexPattern | null>();
+  const [geoFields, setGeoFields] = useState<IndexPatternField[]>();
+  const [selectedField, setSelectedField] = useState<IndexPatternField | null | undefined>();
+  const [documentRequestNumber, setDocumentRequestNumber] = useState<number>(
+    selectedLayerConfig.source.documentRequestNumber
+  );
+
+  const formatFieldToComboBox = (field?: IndexPatternField | null) => {
+    if (!field) return [];
+    return formatFieldsToComboBox([field]);
+  };
+
+  const formatFieldsToComboBox = (fields?: IndexPatternField[]) => {
+    if (!fields) return [];
+
+    return fields?.map((field) => {
+      return {
+        label: field.displayName || field.name,
+      };
+    });
+  };
+
+  const onDocumentRequestNumberChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    const selectedNumber = parseInt(value, 10);
+    setDocumentRequestNumber(selectedNumber);
+    const source = { ...selectedLayerConfig.source, documentRequestNumber: selectedNumber };
+    setSelectedLayerConfig({ ...selectedLayerConfig, source });
+  };
+
+  useEffect(() => {
+    const selectIndexPattern = async () => {
+      if (selectedLayerConfig.source.indexPatternId) {
+        const savedIndexPattern = await indexPatterns.get(
+          selectedLayerConfig.source.indexPatternId
+        );
+        setIndexPattern(savedIndexPattern);
+      }
+    };
+    selectIndexPattern();
+  }, [indexPatterns]);
+
+  // Update the fields list every time the index pattern is modified.
+  useEffect(() => {
+    const fields = indexPattern?.fields.filter((field) => field.type === 'geo_point');
```

Comment:
shall we move geo_point to accepted_field_types list? hence, we can add geo_shape in next PR.

Reply:
good idea, updated.